android-kotlin/app/src/main/java/com/mongodb/app/TaskAdapter.kt

```diff
@@ -0,0 +1,75 @@
+package com.mongodb.app
+
+import android.view.*
+import android.widget.CheckBox
+import android.widget.PopupMenu
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+import io.realm.OrderedRealmCollection
+import io.realm.Realm
+import io.realm.RealmConfiguration
+import io.realm.RealmRecyclerViewAdapter
+import io.realm.kotlin.where
+import org.bson.types.ObjectId
+
+/**
+ * Extends the Realm-provided RealmRecyclerViewAdapter to provide data
+ * for a RecyclerView to display Realm objects on screen to a user.
+ */
+internal class TaskAdapter(data: OrderedRealmCollection<Task>, private val config: RealmConfiguration) : RealmRecyclerViewAdapter<Task, TaskAdapter.TaskViewHolder?>(data, true) {
+
+    internal inner class TaskViewHolder(view: View) : RecyclerView.ViewHolder(view) {
+        var id: ObjectId? = null
+        var name: TextView = view.findViewById(R.id.name)
+        var menu: TextView = view.findViewById(R.id.popup_menu)
+        var checkbox: CheckBox = view.findViewById(R.id.checkbox)
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): TaskViewHolder {
+        val taskView = LayoutInflater.from(parent.context).inflate(R.layout.task_view, parent, false)
+        return TaskViewHolder(taskView)
+    }
+
+    override fun onBindViewHolder(holder: TaskViewHolder, position: Int) {
+        val task: Task = getItem(position) ?: return
+        holder.id = task._id
+        holder.name.text = task.summary
+        holder.checkbox.isChecked = task.isComplete
+        holder.checkbox.setOnClickListener { onCheckboxClicked(holder) }
+        holder.menu.setOnClickListener { onMenuClicked(holder) }
+    }
+
+    /**
+     *  Allows a user to check and uncheck a task and updates its status in the realm.
+     */
+    private fun onCheckboxClicked(holder: TaskViewHolder) {
+        val realm: Realm = Realm.getInstance(config)
```

Comment:
There shouldn't be a need to open the Realm here... you can get it from the RealmResults using `results.getRealm()`

Reply:
I don't think this is true since the adapter actually accepts an OrderedRealmCollection as an argument, and that class doesn't have a `getRealm()` method that I can see. We could always accept a RealmResults instead, but then folks can't use other implementations of OrderedRealmCollection... considering the fact that initializing a realm that's already open is quite cheap, I think we might just have to accept this one.